modules/apigee/main.tf

```diff
@@ -23,11 +23,12 @@ resource "google_apigee_organization" "organization" {
   count                                = var.organization == null ? 0 : 1
   analytics_region                     = var.organization.analytics_region
   project_id                           = var.project_id
-  authorized_network                   = var.organization.authorized_network
+  authorized_network                   = var.organization.disable_vpc_peering ? null : var.organization.authorized_network
```

Comment:
Instead of changing the value here I'd add validation in the variable ensuring `var.organization.authorized_network = null` when `var.organization.disable_vpc_peering` is truee